disperser/controller/controller_config.go

```diff
@@ -0,0 +1,143 @@
+type ControllerConfig struct {
+	// PullInterval is how frequently the Dispatcher polls for new encoded blobs to batch and dispatch.
+	// Must be positive.
+	PullInterval time.Duration
+
+	// DisperserID is the unique identifier for this disperser instance.
+	DisperserID uint32
+
+	// FinalizationBlockDelay is the number of blocks to wait before using operator state.
+	// This provides a hedge against chain reorganizations.
+	FinalizationBlockDelay uint64
+
+	// BatchMetadataUpdatePeriod is the interval between attempts to refresh batch metadata
+	// (reference block number and operator state).
+	// Since this changes at most once per eth block, values shorter than 10 seconds are not useful.
+	// In practice, checking every several minutes is sufficient.
+	// Must be positive.
+	BatchMetadataUpdatePeriod time.Duration
+
+	// AttestationTimeout is the maximum time to wait for a single node to provide a signature.
+	// Must be positive.
+	AttestationTimeout time.Duration
+
+	// BatchAttestationTimeout is the maximum time to wait for all nodes to provide signatures for a batch.
+	// Must be positive and must be longer or equal to the AttestationTimeout.
+	BatchAttestationTimeout time.Duration
+
+	// SignatureTickInterval is how frequently attestations are updated in the blob metadata store
+	// as signature gathering progresses.
+	// Must be positive.
+	SignatureTickInterval time.Duration
+
+	// MaxBatchSize is the maximum number of blobs to include in a single batch for dispersal.
+	// Must be at least 1.
+	MaxBatchSize int32
+
+	// SignificantSigningThresholdFraction is a configurable "important" signing threshold fraction.
+	// Used to track signing metrics and understand system performance.
+	// If the value is 0, special handling for this threshold is disabled.
+	// Must be between 0.0 and 1.0.
+	SignificantSigningThresholdFraction float64
+
+	// Whether or not to collect detailed validator signing metrics.
+	CollectDetailedValidatorSigningMetrics bool
+
+	// NumConcurrentRequests is the size of the worker pool for processing dispersal requests concurrently.
+	// Must be at least 1.
+	NumConcurrentRequests int
+
+	// NodeClientCacheSize is the maximum number of node clients to cache for reuse.
+	// Must be at least 1.
+	NodeClientCacheSize int
+
+	// MaxDispersalAge is the maximum age a dispersal request can be before it is discarded.
+	// Dispersals older than this duration are marked as Failed and not processed.
+	//
+	// Age is determined by the BlobHeader.PaymentMetadata.Timestamp field, which is set by the
+	// client at dispersal request creation time (in nanoseconds since Unix epoch).
+	MaxDispersalAge time.Duration
+
+	// The amount of time to retain signing rate data.
+	SigningRateRetentionPeriod time.Duration
+
+	// The duration of each signing rate bucket. Smaller buckets yield more granular data, at the cost of memory
+	// and storage overhead.
+	SigningRateBucketSpan time.Duration
```

Comment:
Everything except for these last two config values are copy-pasted from the old location